### ISSUE TITLE
How to self-host API?

### ISSUE THREAD

**Issue by Nyankoo** [WeDontPanic/Jotoba#46]
With the website down until August, I was wondering how to host the API on my own server and where I can get the files to do so?

Any help would be greatly appreciated!

**Comment by ElnuDev** [comment]
I'm planning on trying to get that set up later today, I'll get back to you on how that goes.

**Comment by ElnuDev** [comment]
On Ubuntu the compilation dependencies are `libleptonica-dev`, `libclang-dev`, `libtesseract-dev`, and `cmake`. Unfortunately, there are some resource files that are required and are seemingly unavailable, looking at #3 and #10. Maybe I'm missing something, but it's rather disappointing considering the repo description says Jotoba is self-hostable. :confused: Hopefully they're made available soon.

**Comment by Nyankoo** [comment]
@ElnuDev Could you share the link to the instructions on how to install the server (when there is one)?

**Comment by ElnuDev** [comment]
It seems the required resource files and tooling required to generate them are not pubic, so Jotoba isn't possible to self-host. I've contacted the devs and they said making it self-hostable is "currently out of scope," so I'm not sure when/if they'll be made available.

To get the server installed (albeit not functioning without the required resources), assuming you're on Debian (I tested on Ubuntu):

1. [Install Rust](https://www.rust-lang.org/tools/install): `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
2. Install the required compile dependencies: `sudo apt install libleptonica-dev libclang-dev libtesseract-dev cmake`
3. Clone the repository and enter it: `git clone https://github.com/WeDontPanic/Jotoba && cd Jotoba`
4. Run: `cargo run -- -s`

At this point it'll compile and start up, but fail to actually start Jotoba due to the lack of resource files.
